src/main/java/com/fjb/sunrise/repositories/UserRepository.java

```diff
@@ -2,9 +2,14 @@
 
 import com.fjb.sunrise.models.User;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface UserRepository extends JpaRepository<User, Long> {
     boolean existsByUsername(String username);
 
     User findByUsername(String username);
+
+    @Query("FROM User u WHERE u.email = :username OR u.phone = :username")
+    User findByEmailOrPhone(@Param("username") String username);
```

Comment:
consider user only 1 method with 2 parameter to check in the same time

Reply:
i will use it in AuthController, i dont want to delete your method